packages/tailwindcss/src/canonicalize-candidates.ts

```diff
@@ -48,23 +73,49 @@ export function createSignatureOptions(
   designSystem: DesignSystem,
   options?: CanonicalizeOptions,
 ): SignatureOptions {
-  return optionsCache.get(designSystem).get(options?.rem ?? null)
+  return signatureOptionsCache.get(designSystem).get(options?.rem ?? null)
+}
+
+const internalOptionsCache = new DefaultMap((designSystem: DesignSystem) => {
+  return new DefaultMap((signatureOptions: SignatureOptions) => {
+    return new DefaultMap((features: Features) => {
```

Comment:
The public API is just an object, but internally we can use a number which makes it easier to use as a cache key. One day we will have Record / Tuple types, but not yet...